app/src/main/java/it/ministerodellasalute/immuni/ui/upload/UploadDataFragment.kt

```diff
@@ -128,7 +135,9 @@ class UploadDataFragment : Fragment(R.layout.upload_data_fragment), Confirmation
             activity?.finish()
         }
         if (requestCode == ALERT_REQUEST_ERROR) {
-            uploadOtp()
+            GlobalScope.launch {
```

Comment:
Is there a specific reason why you introduced this `GlobalScope` non lifecycle aware usage?